server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -734,6 +736,53 @@ private static void validateDataStreamsStillReferenced(ClusterState state, Strin
         }
     }
 
+    // This method should be invoked after validateDataStreamsStillReferenced(...)
```

Comment:
Is the reason why it should be invoked after because we want to check the more generic case first before the more specific TSDB case?

Reply:
Yes, currently this is mainly because it assumes that templates that do match with existing data streams have a `DataStreamTemplate`, which is something that `validateDataStreamsStillReferenced(...)` is checking for.